### PR TITLE
PlaylistMaker Sprint_19 v1.0

### PR DESCRIPTION
**Pull request by AlexKutsyk**

### CODE REVIEW

File: app/src/main/java/com/practicum/playlistmaker/library/ui/FavoritesFragment.kt [hunk 1]
```diff
@@ -6,6 +6,10 @@ import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import com.practicum.playlistmaker.databinding.FragmentFavoritesBinding
+<<<<<<< HEAD
```
Comment (ArturNurtdinov) [inline app/src/main/java/com/practicum/playlistmaker/library/ui/FavoritesFragment.kt]: Тут мердж конфликт который нужно решить

File: app/src/main/java/com/practicum/playlistmaker/library/ui/LibraryAdapter.kt [hunk 2]
```diff
@@ -13,6 +13,6 @@ class LibraryAdapter(
     }
 
     override fun createFragment(position: Int): Fragment {
-        return if (position == 0) FavoritesFragment.newInstance() else PlaylistsFragment.newInstance()
+        return if (position == 0) FavoritesFragment() else PlaylistsFragment()
```
Comment (ArturNurtdinov) [inline app/src/main/java/com/practicum/playlistmaker/library/ui/LibraryAdapter.kt]: Здесь нужно вернуть создание через `newInstance`, это есть в требованиях

File: app/src/main/java/com/practicum/playlistmaker/library/ui/LibraryFragment.kt [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+                1 -> tab.text = getString(R.string.playlists)
+            }
+        }
+
+        tabMediator.attach()
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        _binding = null
```
Comment (ArturNurtdinov) [inline app/src/main/java/com/practicum/playlistmaker/library/ui/LibraryFragment.kt:45]: Здесь дополнительно нужно ещё `tabMediator.detach()` вызвать, чтоб не было утечки

File: app/src/main/java/com/practicum/playlistmaker/player/ui/PlayerActivity.kt [hunk 4]
```diff
@@ -184,6 +177,26 @@ class PlayerActivity : AppCompatActivity() {
...
+                timePlaying.text = formatTimeTrack.format(START_TIME)
+                yearValue.text = formatYearTrack.format(
+                    SimpleDateFormat(getString(R.string.year_track_yyyy)).parse(track.releaseDate)
+                )
+            }
+        }
+    }
+
+    private fun showScreenPlayerWODemo() {
+        binding.timePlaying.text = "No Demo"
```
Comment (ArturNurtdinov) [inline app/src/main/java/com/practicum/playlistmaker/player/ui/PlayerActivity.kt:197]: Все строчки, которые видны пользователю, необходимо выносить в strings.xml

File: app/src/main/java/com/practicum/playlistmaker/search/domain/models/Track.kt [hunk 5]
```diff
@@ -13,7 +13,7 @@ data class Track(
     val releaseDate: String?,
     val primaryGenreName: String?,
     val country: String?,
-    val previewUrl: String?,
+    var previewUrl: String?,
```
Comment (ArturNurtdinov) [inline app/src/main/java/com/practicum/playlistmaker/search/domain/models/Track.kt:16]: Лучше оставить val, на будущее - всегда лучше в domain моделях все поля делать имьютабельными (val)

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -25,6 +25,8 @@
     <string name="time_track_mm_ss">mm:ss</string>
     <string name="year_track_yyyy">yyyy</string>
     <string name="no_demo">No Demo</string>
+    <!-- TODO: Remove or change this placeholder text -->
+    <string name="hello_blank_fragment">Hello blank fragment</string>
```
Comment (ArturNurtdinov) [inline app/src/main/res/values/strings.xml:29]: можно удалить это
